File: PHP/linearized-pdf-endpoint.php
<?php
require("../Sample_Input/sample_input.php");

// The /linearized-pdf endpoint can take a single PDF file or id as input.
// This sample demonstrates linearizing a PDF file.
$linearized_pdf_endpoint_url = 'https://api.pdfrest.com/linearized-pdf';

// Create an array that contains that data that will be passed to the POST request.
$data = array(
    'file' => new CURLFile(SAMPLE_INPUT_DIR . 'toLinearize.pdf','application/pdf', 'toLinearize.pdf'),
    'output' => 'example_linearizedPdf_out'
);

$headers = array(
    'Accept: application/json',
    'Content-Type: multipart/form-data',
    'Api-Key: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' // place your api key here
);

// Initialize a cURL session.
$ch = curl_init();

// Set the url, headers, and data that will be sent to linearized-pdf endpoint.
curl_setopt($ch, CURLOPT_URL, $linearized_pdf_endpoint_url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, $data);

print "Sending POST request to linearized-pdf endpoint...\n";
$response = curl_exec($ch);

print "Response status code: " . curl_getinfo($ch, CURLINFO_HTTP_CODE) . "\n";

if($response === false){
    print 'Error: ' . curl_error($ch) . "\n";
}else{
    print json_encode(json_decode($response), JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
    print "\n";
}

curl_close($ch);
?>

File: Python/linearized-pdf-endpoint.py
import requests

# Set the API endpoint URL
url = "https://api.pdfrest.com/linearized-pdf"

# Define the payload data for the request
payload = {
    'output': 'pdfrest_linearized_pdf'
}

# Specify the PDF file to be linearized
files = [
    ('file', ('file', open('/path/to/file', 'rb'), 'application/octet-stream'))
]

# Set the headers, including the API key
headers = {
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'  # Replace with your API key
}

# Make a POST request to the API endpoint
response = requests.request(
    "POST", url, headers=headers, data=payload, files=files)

# Print the response content
print(response.text)
